metricflow/model/semantics/linkable_spec_resolver.py

```diff
@@ -627,13 +669,24 @@ def _get_metric_time_elements(self, measure_reference: MeasureReference) -> Link
         )
 
         # For each of the possible time granularities, create a LinkableDimension for each one.
-        return LinkableElementSet(
-            path_key_to_linkable_dimensions={
-                ElementPathKey(
+
+        path_key_to_linkable_dimensions: Dict[ElementPathKey, List[LinkableDimension]] = defaultdict(list)
+        for time_granularity in possible_metric_time_granularities:
+            possible_date_parts: Sequence[Optional[DatePart]] = (
+                # No date part, just the metric time at a different grain.
+                (None,)
+                # date part of a metric time at a different grain.
+                + tuple(date_part for date_part in DatePart if time_granularity.to_int() <= date_part.to_int())
+            )
```

Comment:
nit: I find this hard to read because it looks like a tuple of tuples and that leading `+` gets elided a bit by the commentary. Maybe it's a "Tom, you need reading glasses" problem, but there it is. There are lots of ways to restructure this so it's easier to see. Here's one:

```
# No date part (i.e., just metric time) plus all date parts for metric time at a different grain
possible_date_parts = (None,) + tuple([date_part for date_part....])
```

In theory the typechecker should infer that correctly but if it doesn't it's more contained and easier to reorient. Note you can also keep the multi-line commentary by using locals and just doing the assignment at the end, whatever works.

Reply:
Will update before merging.